File: Exercise 1/app/src/main/java/com/example/recycle_view/myAdapter.java
package com.example.recycle_view;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class myAdapter extends RecyclerView.Adapter<myAdapter.MyViewHolder> {
    private static final String TAG = "myAdapter";
    private ArrayList<MyData> myData;
    private Context context;

    myAdapter(Context context, ArrayList<MyData> myData){
        this.context = context;
        this.myData = myData;
    }


    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Log.i(TAG, "onCreateViewHolder");
        /*LinearLayout linearLayout = new LinearLayout(context);
        linearLayout.setLayoutParams(new ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
        linearLayout.setOrientation(LinearLayout.VERTICAL);
        // Create a stand-alone view
        View myView = LayoutInflater.from(context)
                .inflate(R.layout.mylayout, null, false);
        linearLayout.addView(myView);*/
        View myView = LayoutInflater.from(parent.getContext()).inflate(R.layout.mylayout, parent, false);
        MyViewHolder vh = new MyViewHolder(myView);
        return  vh;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        Log.i(TAG, "onBindViewHolder");
        final MyData m = myData.get(position);
        holder.textview.setText(myData.get(position).getName());
    }

    @Override
    public int getItemCount() {
        Log.i(TAG, "getItemCount");
        return myData.size();
    }


     static class MyViewHolder extends RecyclerView.ViewHolder {
        TextView textview;

         MyViewHolder(@NonNull View itemView) {
            super(itemView);
             Log.i(TAG, "MyViewHolder");
            textview = itemView.findViewById(R.id.name);
        }
    }
}

File: exercise11androidintegrationwithwebservices/src/main/java/com/example/exercise11androidintegrationwithwebservices/MyAdapter.kt
package com.example.exercise11androidintegrationwithwebservices

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.exercise11androidintegrationwithwebservices.MyAdapter.MyViewHolder
import kotlinx.android.synthetic.main.mylayout.view.*

class MyAdapter internal constructor(private val myData: List<Posts>) :
    RecyclerView.Adapter<MyViewHolder>() {

    private companion object {
        private const val TAG = "myAdapter"
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val myView: View =
            LayoutInflater.from(parent.context).inflate(R.layout.mylayout, parent, false)
        return MyViewHolder(myView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.dataBind(myData[position])
    }

    override fun getItemCount(): Int {
        return myData.size
    }

    /**
     * ViewHolder Class
     */
    class MyViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        fun dataBind(myData: Posts) {
            itemView.nameID.text = myData.name
            itemView.messageID.text = myData.message
            Glide.with(itemView.context).load(myData.profileImage)
                .into(itemView.photoID)
        }
    }
}